### DISCUSSION TITLE
Beginner  - Search good usage and settings for TPROC-C tests with Posgresql database . I'm stuck in testing with the GUI.

### DISCUSSION THREAD

**Discussion by nomade33** [TPC-Council/HammerDB#338] [Q&A] [answered]
(Sorry for my level of English, I will try to do my best during our exchanges.
Thank you for your understanding)

My goal is to try and scale up gradually through the Hammerdb GUI menu in order to measure the performance 
of Postgresql database installed on a remote linux server with 2 CPUs and 4Gb of RAM.
I try to collect TPS performance with the execution of TPROC-C schema 
from hammerdb 4-4  installed on my Windows workstation with 8 CPUs and 16 GB of RAM.

- Here is my test since the creation of the schema
Benchmark TPROC-C
 - Schema build with following options :
		Number of Warehouse= 10
		Virtual Usersto Build Schema 2
		Prefer PostgreSQL SSL Mode (box checked)
 - Driver Script  is configured with following parameters:
 		Timed driver scrip ( box checked)
		Total Transactions per user:  1000000
		Minute of Rampup = 0
		Minutes of test duration = 3
 
 - I RUN virtual user wit folowing option: 
  Virtuals user 1
  User Delay(ms): 500
  Repeat Delay(ms): 500
  Iteration : 2

 Cannot exceed one iteration for one virtual user
 Output Status is always with a red  tick after the test duration. 

Output message is already when i try to raise the number of iteration :
-  "Warning : a running Virtual User was terminated, any pending output has been discarded. 

Could you please,help me find good scaling parameters for this postgresql test.

Postgresql 13.3 is configured with max_connections=200 and i try to reach and to simulate this load of sessions from HammerDB.

**Comment by sm-shaw** [comment, marked as answer]
The Documentation, in particular the [quick start](https://www.hammerdb.com/docs/ch02.html), can help you with the workflow. 
So to break it down step by step, with the GUI try sharing screenshots. 
Firstly, you need to build a schema, 10 to 20 warehouses is fine for your 2 CPU configuration. Did this work, did you see something as follows? 

![pgschbuild](https://user-images.githubusercontent.com/38044085/162781950-b70a0c10-8062-4832-8152-96fdd2cbf060.png)

Then for the test phase for one virtual user it will look as follows (example from a laptop so not a server configuration) 

![pgdrv](https://user-images.githubusercontent.com/38044085/162782086-2f4b7e03-e3ae-431a-b535-fbbd85813fab.png)

Do not worry about Virtual User iterations, this is for special case testing.  Also with 2 CPUs 200 Virtual Users would be highly over configured for the default CPU intensive workload. Test with 1, 2, 4, 8 - you will probably peak with performance somewhere in this range. 
If you have trouble connecting between systems make sure the pg_hba.conf is configured correctly.

**Reply by nomade33** [reply]
>Hello,

thank you for this first feedback.

I sent you an email with a file including snapshots of my test. I hope you received it.
This is exactly what the documentation describes.
- Creation of the schema with 10 virtual users and 50 Warhouse. Ends with an OK status (green tick in the console)
- Creation of the "driver script" (Timed script)

Then running the test is simple:
- A Virtual user and 1 iteration, it works, the status is green in the HammerDB console but with 1 virtual user for 2 iterations. The test does not work The status is green for the second virtual user and red for the first (MONITOR) with the output message "Warning a running Virtual User was terminated, any pending output has been discarded ... "

If Hammerdb works with "test driver script" mode it works
 for one virtual user and two iterations. 

The only difference between the two tests is the activation of the "Timed script" mode and the presence of an additional MONITORING virtual userwhen i run virtual user.

Also, note that the postgresql database is not installed locally but on a remote server.
Could you guide me in the diagnosis to be made ?

Bests regards

**Reply by sm-shaw** [reply]
Hi,
Support is not provided through email - support is detailed here https://www.hammerdb.com/support.html
If you see the error message "Warning a running Virtual User was terminated, any pending output has been discarded ..." you terminated the virtual user before it completed so you didn't get any output. 
It would be best to follow the documentation carefully this can guide you how to start.
